vars/runGradleCheck.groovy

```diff
@@ -0,0 +1,61 @@
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    def git_repo_url = args.gitRepoUrl ?: 'null'
+    def git_reference = args.gitReference ?: 'null'
+
+    println("Git Repo: ${git_repo_url}")
+    println("Git Reference: ${git_reference}")
+
+    if (git_repo_url.equals('null') || git_reference.equals('null')) {
+        println("No git repo url or git reference to checkout the commit, exit 1")
+        System.exit(1)
+    }
+    else {
+        withCredentials([
+            usernamePassword(credentialsId: "jenkins-gradle-check-s3-aws-credentials", usernameVariable: 'amazon_s3_access_key', passwordVariable: 'amazon_s3_secret_key'),
+            usernamePassword(credentialsId: "jenkins-gradle-check-s3-aws-resources", usernameVariable: 'amazon_s3_base_path', passwordVariable: 'amazon_s3_bucket')]) {
+
+            sh """
+
+                set -e
+                set +x
+
+                env | grep JAVA | grep HOME
+
+                echo "Git clone: ${git_repo_url} with ref: ${git_reference}"
+                rm -rf search
```

Comment:
Do we need to have `search` directory?

Reply:
In this instance, yes and it is better to not mix build repo content with opensearch core content.

Reply:
Build will be placed in a separate directory and rest of the core code will be present separately

Reply:
The other way I can think of is to have a directory name `opensearch` or `opensearch-core` to be more specific.

Reply:
It is not. Jenkins will pull build repo directly to the workspace without folder.

Reply:
As for search folder renaming that can be a follow up improvement PR.